### PR TITLE
Respect `temporary-directory` config for spilling

### PR DESCRIPTION
**Pull request by jakirkham**
Make sure to check `dask.config`'s value for `temporary-directory` when creating a `DeviceHostFile` if `local_directory` is not otherwise specified.

### CODE REVIEW

**Comment by raydouglass** [comment]
# [Codecov](https://codecov.io/gh/rapidsai/dask-cuda/pull/247?src=pr&el=h1) Report
> Merging [#247](https://codecov.io/gh/rapidsai/dask-cuda/pull/247?src=pr&el=desc) into [branch-0.13](https://codecov.io/gh/rapidsai/dask-cuda/commit/dd18b6fa85cfaa6acdb3c53b4cf61c423da5eee8&el=desc) will **decrease** coverage by `0.18%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/rapidsai/dask-cuda/pull/247?src=pr&el=tree)

```diff
@@               Coverage Diff               @@
##           branch-0.13     #247      +/-   ##
===============================================
- Coverage        76.29%   76.11%   -0.19%     
===============================================
  Files               14       14              
  Lines              983      988       +5     
===============================================
+ Hits               750      752       +2     
- Misses             233      236       +3     
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/rapidsai/dask-cuda/pull/247?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/rapidsai/dask-cuda/pull/247?src=pr&el=footer). Last update [dd18b6f...69e24ed](https://codecov.io/gh/rapidsai/dask-cuda/pull/247?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: dask_cuda/tests/test_device_host_file.py [hunk 1]
```diff
@@ -14,6 +15,13 @@
 cupy = pytest.importorskip("cupy")
 
 
+def test_device_host_file_config(tmp_path):
+    dhf_disk_path = str(tmp_path / "dask-worker-space" / "storage")
+    with dask.config.set(temporary_directory=str(tmp_path)):
+        dhf = DeviceHostFile()
+        assert dhf.disk_func_path == dhf_disk_path
```
Comment (pentschev) [inline dask_cuda/tests/test_device_host_file.py]: Wouldn't it make more sense to check if the directory indeed got created? We could also avoid the need to make `disk_func_path` an attribute of `DeviceHostFile`.

Comment (jakirkham) [reply dask_cuda/tests/test_device_host_file.py]: That seems like a good check to add. Happy to add that.

That said, it still seems worthwhile to verify `DeviceHostFile` is using that path.

**Comment by jakirkham** [comment]
Thanks Peter 🙂

**Pull request merged**